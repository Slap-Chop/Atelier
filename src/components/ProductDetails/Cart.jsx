import React from 'react';
import Select from 'react-select';
import axios from 'axios';
import config from '../../../config.js';

class Cart extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentSize: 'Select Size',
      stock: {},
      inStock: false,
      stockMessage: 'Out of Stock',
      quantity: 0,
      skuId: 0,
      skuObj: {},
      notSelected: false,
    }
    this.sizeRef = React.createRef();
  }

  componentDidUpdate() {
    // console.log(this.state.stock)
    if(this.props.currentStyle.skus !== this.state.stock) {
      this.setState({stock: this.props.currentStyle?.skus, currentSize: 'Select Size'})
      if (this.props.currentStyle?.skus?.length !== 0) {
        this.setState({stockMessage: 'Select Size', inStock: true});
      }
      if (this.props.currentStyle?.skus?.null) {
        this.setState({stockMessage: 'Out of Stock', inStock: false, currentSize: 'Select Size'})
      }
    }
  }

  handleSizeChange(event) {
    let skuId = event?.value
    let skuObj = this.state.stock[skuId] || {quantity: 0, size: 'Select Size'}
    this.setState({notSelected: false});
    // setting state to have the select sku information
    if (skuId === 'N/A') {
      this.setState({sku: 0,
      skuObj: {},
      currentSize: 'Select Size',
      quantity: 0,
      })
    } else {
      this.setState({sku: skuId,
        skuObj: skuObj,
        currentSize: skuObj?.size,
        quantity: this.state.quantity > skuObj.quantity ? 1 : this.state.quantity ? this.state.quantity : 1
      })
    }
    // console.log(skuId, this.state.stock[skuId], skuObj?.size)
  }

  handleQuantChange(event) {
    // console.log(event?.target?.value)
    this.setState({quantity: event?.target?.value})
  }

  handleAddCart() {
    // console.log(this.props.currentStyle.name, this.state.quantity, this.state.currentSize)
    let productObj = {
      sku_id: this.state.sku,
      count: this.state.quantity
    };
    // console.log(productObj)
    if (this.state.currentSize === 'Select Size') {
      this.setState({notSelected: true})
      this.sizeRef.current?.focus();
    } else {
      axios.defaults.headers.common['Authorization'] = config.TOKEN;
      axios.post('https://app-hrsei-api.herokuapp.com/api/fec2/hr-rfp/cart/', productObj).then((response) => {
        console.log(`product ${this.props.currentStyle.name} added to cart with size ${this.state.currentSize}, quantity ${this.state.quantity}, sku ${this.state.sku} and response status ${response.status}`)
      })
    }
    // console.log('skus', this.props.currentStyle?.skus)

  }

  render() {
    let quantArray = [];
    let maxquant = this.state.skuObj?.quantity > 15 ? 15 : this.state.skuObj?.quantity
    for (let i = 1; i <= maxquant; i++) {
      quantArray.push(i)
    }

    let options = [{value: 'N/A', label: this.state.stockMessage}]

    if (this.state.inStock && this.state.stock) {
      let newoptions = [];
      newoptions.push({value: 'N/A', label: 'Select Size'})
      Object.entries(this.state.stock).forEach((sku, index) => {
        newoptions.push({value: sku[0], label: sku[1].size})
      })
      options = newoptions
    }

    return(
      <div className='productBorder'>
        {this.props.currentProduct?.name} > {this.props.currentStyle?.name}
      <div className='cartContainer'>
      <div className='sizeContainer'>
      {/* Size dropdown menu */}
      {this.state.notSelected && 'Please select a size'}
      {this.state.inStock &&
        <Select onChange={this.handleSizeChange.bind(this)}
      openMenuOnFocus={true}
      defaultValue={{label: 'Select Size', value: 'N/A'}}
      options={options}
      ref={this.sizeRef}
      className='sizeMenu'
      isDisabled={!this.state.inStock}/>}
      {/* check if item is in stock*/}
      {!this.state.inStock && <Select onChange={this.handleSizeChange.bind(this)}
      openMenuOnFocus={true}
      defaultValue={{label: 'Out of Stock', value: 'N/A'}}
      options={options}
      ref={this.sizeRef}
      className='sizeMenu'
      isDisabled={!this.state.inStock}/>}
        {/* <option value={'N/A'}>{this.state.stockMessage}</option> */}
      {/* map and add sizes to the dropdown */}
        {/* {this.state.stock && Object.entries(this.state.stock)?.map((sku, index) => {
        return (<option value={sku[0]} key={index}>{sku[1].size}</option>)
      })} */}


      </div>
      <div>
        {/* Quantity dropdown menu */}
        <select className='quantityMenu'
        onChange={this.handleQuantChange.bind(this)}
        disabled={this.state.currentSize === 'Select Size'}>
        {this.state.currentSize === 'Select Size' && <option>-</option>}
        {this.state.currentSize !== 'Select Size' && quantArray.map((number, index) => {
          return <option key={index} value={number}>{number}</option>
        })}
        </select>
      </div>

      </div>

      {/* Add to Cart Button */}
      {this.state.inStock && <button onClick={this.handleAddCart.bind(this)}>Add to Cart</button>}
    </div>
    )
  }


}

export default Cart;