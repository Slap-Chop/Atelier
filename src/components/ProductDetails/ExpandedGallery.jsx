import React from "react";
import ReactDOM from 'react-dom';
import Xbutton from './Images/Xbutton.png';

class ExpandedGallery extends React.Component {
  constructor(props){
    super(props);
    const overlayStyle = {
      position: 'fixed', /* Stay in place */
      zIndex: 1, /* Sit on top */
      left: 0,
      top: 0,
      width: '100%', /* Full width */
      height: '100%', /* Full height */
      justifyContent: 'center',
      display: 'flex',
      backgroundColor: 'rgb(0,0,0)', /* Fallback color */
      backgroundColor: 'rgba(0,0,0,0.4)' /* Black w/ opacity */
    }
  }



  render() {
    if (!this.props.open) {
      return null;
    }
    return (
    <div className='modal' style ={{
      position: 'fixed',
      zIndex: 1,
      left: 0,
      top: 0,
      width: '100%',
      height: '100%',
      justifyContent: 'center',
      display: 'flex',
      backgroundColor: 'rgb(0,0,0)',
      backgroundColor: 'rgba(0,0,0,0.4)'
    }}>
      <div className='modal-content' style={{
        width: '90%',
        height: '90%',
        backgroundImage:`url(${this.props.photos[this.props.photoIndex]?.url})`,
        backgroundRepeat: 'no-repeat',
        backgroundPosition: 'center',
        backgroundSize: 'contain',
      }}>
        <div className='modal-header'>
        <img
        src={Xbutton}
        onClick={(e) => this.props.xClick && this.props.xClick(e)}
        style={{
          width: '50px',
          height: '50px',
          display: 'flex',
          textAlign: 'center',
          textAlignVertical: 'center',
          paddingLeft: '5px',
          }}/></div>
        <div className='modal-body'></div>

      </div>
    </div>
    )
  }
}


export default ExpandedGallery;