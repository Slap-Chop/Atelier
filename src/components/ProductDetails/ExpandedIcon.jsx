import React from 'react';

const ExpandedIcon = (props) => {
  return (
    <div
    className='expandedIconContainer'
      >
      <img
        onClick={() => {
          props.click(props.index)
        }}
        className='expandedIcon'
        style={{opacity: props.selected}}
      src={`${props.photo.thumbnail_url}`}/>

    </div>
  )
}

export default ExpandedIcon;