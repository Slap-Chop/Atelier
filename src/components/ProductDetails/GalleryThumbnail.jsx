import React from 'react';
import noImage from './Images/noImage.png';

const GalleryThumbnail = (props) => {
  return (
    <div
    className='galleryThumbnailContainer'
      >
      <img
        className='galleryThumbnail'
        onClick={() => {
          props.click(props.index)
        }}
        style={{opacity: props.selected}}
      src={`${props.photo.thumbnail_url}` || noImage}
      alt='Image N/A'/>

    </div>
  )
}

export default GalleryThumbnail;