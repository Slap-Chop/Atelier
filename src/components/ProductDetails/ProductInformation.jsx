import React from 'react';

const ProductInfo = (props) => {
  console.log('currentProduct:', props.currentProduct)
  return (
  <div style={{border: '1px solid blue'}}>
    Review star component
    <h2><p>{props.currentProduct.category}</p> {props.currentProduct.name}</h2>
    <p>${props.currentProduct.default_price}</p>
    <p>Product Overview, sometimes exists</p>
    <p>social media buttons</p>
  </div>
)}

export default ProductInfo;