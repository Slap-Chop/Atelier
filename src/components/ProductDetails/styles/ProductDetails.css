.productDetailsBody {
  border: 1px solid black;
  display: flex;
  justify-content: center;
}

.flex {
  display: flex;
}

.imageGalleryBody {
  border: 1px solid black;
  width: 130%;
}

.productInfoBody {
  float: right;
  width: 95%;
  border: 1px solid black;
}

/* .productBorder {
  border: 1px solid black;
} */

.imageGalleryContainer {
  position: relative;
  background-color: lightgrey;
}

.imageContainer {
  display: flex;
  justify-content: center;
}

.mainImage {
  position: absolute;
  display: flex;
  width: 100%;
  height: 100%;
  object-fit: contain;
  justify-content: center;
  cursor: zoom-in;
}

.vertArrow {
  width: 50px;
  height: 50px;
  display: flex;
  padding-left: 5px;
  z-index: 3;
  position: relative;
  object-fit: contain;
}

.expandedUp {
  top: 12%;
  left: 5%;
  position: absolute;
  padding-left: 5px;
}

.expandedDown {
  top: 60%;
  left: 5%;
  position: absolute;
  padding-left: 5px;
}

.leftArrowContainer {
  position: absolute;
  top: 50%;
  left: 10%;
}

.rightArrowContainer {
  position: absolute;
  top: 50%;
  right: 0%;
}

.horzArrow{
  background: rgba(105, 105, 105, .5);
  width: 25px;
  height: 60px;
  object-fit: cover;
  z-index: 3;
}

.productName {
  font-size: 2em;
  font-weight: 600;
}

.salePrice{
  padding-left: 10px;
}

.styleComponentContainer{
  width: 75px;
  height: 75px;
  border-radius: 100%;
  display: flex;
  padding: 5px;
}

.styleThumbnail{
  float: right;
  transform-origin: 50% 50%;
  border-radius: 50%;
  height: 50px;
  width: 50px;
  object-fit: cover;
}

.thumbnailCheck {
  width: 15px;
  height: 15px;
  background-color:white;
  border-radius: 50%;
  object-fit: cover;
}

.galleryThumbnailContainer {
  width: 100px;
  height: 100px;
  display: flex;
  padding: 1px;
}

.galleryThumbnail {
  float: right;
  transform-origin: 50% 50%;
  border-radius: 10%;
  height: 60px;
  width: 60px;
  object-fit: cover;
  opacity: .5;
  z-index: 3;
}

.expandedIconContainer {
  width: 50px;
  height: 50px;
  display: flex;
  padding: 1px;
}

.expandedIcon {
  float: right;
  transform-origin: 50% 50%;
  border-radius: 50%;
  height: 40px;
  width: 40px;
  object-fit: cover;
  opacity:.5;
  z-index: 3;
}

.cartContainer {
  display: flex;
  padding-bottom: 20px;
}

.sizeContainer{
  padding-right: 30px;
}

.sizeMenu {
  width: 200px;
  height: 40px;
}

.quantityMenu {
  width: 80px;
  height: 40px;
}

.modal {
  position: fixed;
  z-index: 1;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  justify-content: center;
  display: flex;
  background-color: rgb(0,0,0);
  background-color: rgba(0,0,0,0.4);
}

.modalContent {
  width: 90%;
  height: 90%;
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
}

.modalImageContainer {
  justify-content: center;
  display: flex;
}

.modalImage {
  position: absolute;
  display: flex;
  max-width: 100%;
  max-height: 100%;
  object-fit: contain;
  justify-content: center;
  cursor: zoom-in;
}

.zoomedImage {
  transform: scale(2.5);
  cursor: zoom-out;
}

.XButton {
  position: absolute;
  top: 3%;
  left: 5%;
  width: 50px;
  height: 50px;
  display: flex;
  padding-left: 5px;
  z-index: 3;
}

.iconContainer {
  position: absolute;
  top: 18%;
  left: 5.5%;
}