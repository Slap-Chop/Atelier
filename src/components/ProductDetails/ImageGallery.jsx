import React from 'react';
import GalleryThumbnail from './GalleryThumbnail.jsx';
import leftArrow from './Images/leftArrow.png';
import rightArrow from './Images/rightArrow.png';
import upArrow from './Images/upArrow.png';
import downArrow from './Images/downArrow.png';
import ExpandedGallery from './ExpandedGallery.jsx';
import noImage from './Images/noImage.png';

class ImageGallery extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      expanded: false,
      photos: [],
      photoIndex: 0,
      thumbnails: [],
      offset: 0,
      LRClick: false,
      expanded: false
    }
  }

  componentDidMount() {
    this.setState({
      photos: this.props.style.photos})
    if (this.props.style.photos?.length > 7) {
      let array = this.props.style.photos?.slice(0 + this.state.offset, 7 + this.state.offset);
      this.setState({thumbnails: array})
    } else {
      this.setState({thumbnails: this.props.style.photos})
    }
  }

  componentDidUpdate(prevProps, prevState) {
    // console.log(this.state.photos, this.state.thumbnails)
    //make sure photo list updates accordingly to props
    if (this.state.photos !== this.props.style.photos && this.props.style.photos) {
      this.setState({photos:this.props.style.photos})
      // cut the 7 thumbnail images out
      if (this.props.style.photos.length > 7) {
        let thumbArray = this.props.style.photos.slice(0 + this.state.offset, 7 + this.state.offset);
        this.setState({thumbnails: thumbArray})
      } else {
        this.setState({thumbnails: this.props.style.photos})
      }
    }
    //reset photoindex if product changes to not index into undefined
    if (this.props.id !== prevProps.id) {
      this.setState({photoIndex: 0})
    }
    // update thumbnail list if the offset changes
    if (this.state.offset !== prevState.offset) {
      let thumbArray = this.props.style.photos.slice(0 + this.state.offset, 7 + this.state.offset);
      this.setState({thumbnails: thumbArray})
    }
    //snap the offset(thumbnail list) to the current picture if the left right arrow goes to something out of range
    if (this.state.photoIndex > this.state.offset + 6 && this.state.LRClick) {
      this.setState({offset: this.state.photoIndex - 6, LRClick: false})
    }
    if (this.state.photoIndex < this.state.offset && this.state.LRClick) {
      this.setState({offset: this.state.photoIndex, LRClick: false})
    }
    //if going to index out of bounds, reset
    if (prevState.photoIndex > this.state.photos?.length) {
      this.setState({photoIndex:0, offset: 0})
    }
  }

  toggleExpanded() {
    if (this.state.expanded === false) {
      this.setState({expanded: true})
    }
  }

  closeExpanded() {
    this.setState({expanded: false})
  }

  handleThumbnailClick(index) {
    this.setState({photoIndex: index + this.state.offset})
  }

  handleRightClick() {
    let index = this.state.photoIndex;
    if (index < this.state.photos.length - 1) {
      index++;
      this.setState({photoIndex: index, LRClick: true})
    }
  }

  handleLeftClick() {
    let index = this.state.photoIndex;
    if (index > 0) {
      index--;
      this.setState({photoIndex: index, LRClick: true})
    }
  }

  handleDownClick() {
    let offset = this.state.offset;
    if (offset + 7 < this.state.photos.length) {
      // console.log('clickdown', this.state.offset)
      offset++;
      this.setState({offset: offset})
    }
  }

  handleUpClick() {
    let offset = this.state.offset;
    if (offset > 0) {
      // console.log('clickup', this.state.offset)
      offset--;
      this.setState({offset: offset})
    }
  }

  render() {
    // make sure props have been passed before trying to render
    if (this.state.photos && this.state.photos[this.state.photoIndex]) {
      return(
        <div className='imageGalleryContainer'>
          <div className='imageContainer'>
          {/* main image */}
          <img onClick={this.toggleExpanded.bind(this)}
          src={this.props.style.photos[this.state.photoIndex]?.url || noImage}
          className='mainImage'/>

        </div>

          <div>
            {/* up arrow, hide if expanded */}
            {!this.state.expanded && this.state.photos.length > 7 && this.state.offset > 0 &&
            <img src={upArrow}
              onClick={this.handleUpClick.bind(this)}
              className='vertArrow'/>
            }
            {!this.state.expanded && this.state.thumbnails.map((photo, index) => {
              if (index + this.state.offset === this.state.photoIndex) {
                //return a highlighted component if it is the current photo
                return(
                <GalleryThumbnail
                  click={this.handleThumbnailClick.bind(this)}
                  photo={photo}
                  index={index + this.state.offset}
                  key={index}
                  selected={1}
                  />)
              }
              return(
                <GalleryThumbnail
                  click={this.handleThumbnailClick.bind(this)}
                  photo={photo}
                  index={index}
                  key={index}/>
              )
            })}
            {/* down arrow, hide on expand */}
            {!this.state.expanded && this.state.photos?.length > 7 && this.state.offset + 7 < this.state.photos?.length &&
              <img src={downArrow}
              onClick={this.handleDownClick.bind(this)}
              className='vertArrow'
            />}
          </div>
          {/* left arrow, hide arrow if on first image, hide on expand */}
          { this.state.photoIndex !== 0 && !this.state.expanded &&
            <div
              className='leftArrowContainer'
            onClick={this.handleLeftClick.bind(this)}
          ><img src={leftArrow}
              className='horzArrow'
          /></div>}
          {/* hide right arrow if on last image, hide on expand */}
          { this.state.photoIndex !== this.state.photos.length - 1 && !this.state.expanded &&
            <div className='rightArrowContainer'
            onClick={this.handleRightClick.bind(this)}
          ><img src={rightArrow}
              className='horzArrow'
          /></div>}
        <ExpandedGallery
          open={this.state.expanded}
          xClick={this.closeExpanded.bind(this)}
          photos={this.state.photos}
          photoIndex={this.state.photoIndex}
          thumbnails={this.state.thumbnails}
          offset={this.state.offset}
          LRClick={this.state.LRClick}
          handleLeft={this.handleLeftClick.bind(this)}
          handleRight={this.handleRightClick.bind(this)}
          handleUp={this.handleUpClick.bind(this)}
          handleDown={this.handleDownClick.bind(this)}
          handleThumb={this.handleThumbnailClick.bind(this)}

          />
      </div>
      )
    }

  }


}

export default ImageGallery;