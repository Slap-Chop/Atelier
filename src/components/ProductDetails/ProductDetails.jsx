import React from 'react';
import ReactDOM from 'react-dom';
import ProductInfo from './ProductInformation.jsx';
import ImageGallery from './ImageGallery.jsx';
import Cart from './Cart.jsx';
import axios from 'axios';
import config from '../../../config.js';

class ProductDetails extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      styles: [],
      currentStyle: {},
      id: 0
    }
  }

  componentDidMount() {
    this.state.id = this.props.id
    axios.defaults.headers.common['Authorization'] = config.TOKEN;
    axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hr-rfp/products/${this.state.id}/styles`).then((data) => {
      // console.log('style data in Style selector',data.data.results)
      this.setState({styles: data.data.results, currentStyle:data.data.results[0]})
      data.data.results.forEach((result) => {
        if (result["default?"]) {
          // console.log('enter default')
          this.setState({currentStyle: result})
        }
      })
    })
  }

  componentDidUpdate() {
    if (this.state.id !== this.props.id) {
      this.setState({id: this.props.id});
      axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hr-rfp/products/${this.props.id}/styles`).then((data) => {
      // console.log('style data in Style selector update',data.data.results)
      this.setState({styles: data.data.results, currentStyle:data.data.results[0]})
      data.data.results.forEach((result) => {
        if (result["default?"]) {
          // console.log('enter default')
          this.setState({currentStyle: result})
        }
      })
    })

    }
    // console.log(this.state.currentStyle, this.props.id, this.props.currentProduct)
  }

  handleStyleChange(style) {
    this.setState({currentStyle: style});
  }


  render() {
    return(<div style={{border: '1px solid blue',
    display: 'flex',
    justifyContent: 'center' }}>
      <div style={{
        border: '1px solid red',
        width: '80%'
        }}>
        <ImageGallery style={this.state.currentStyle}/></div>
      <div>
      <ProductInfo currentProduct={this.props.currentProduct}
      styles={this.state.styles}
      currentStyle={this.state.currentStyle}
      click={this.handleStyleChange.bind(this)}
      id={this.props.id}/>
      <Cart currentStyle={this.state.currentStyle}
      currentProduct={this.props.currentProduct}/>
      <p>social media buttons</p>
      <div style={{border: '1px solid green'}}>
        <h5>{this.props.currentProduct.slogan}</h5>
        <p>{this.props.currentProduct.description}</p></div>
      </div>

    </div>

    )
  }
}

export default ProductDetails;