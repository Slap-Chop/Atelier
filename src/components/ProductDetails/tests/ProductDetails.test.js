import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import ProductDetails from '../ProductDetails.jsx';

const testList = [{
  category: "Jackets",
  default_price: "140.00",
  description: "The So Fatigues will wake you up and fit you in. This high energy camo will have you blending in to even the wildest surroundings.",
  id: 40344,
  name: "Camo Onesie",
  slogan: "Blend in to your crowd"
},
{
  category: "Jackets",
  default_price: "140.00",
  description: "The So Fatigues will wake you up and fit you in. This high energy camo will have you blending in to even the wildest surroundings.",
  id: 40345,
  name: "Camo Onesie",
  slogan: "Blend in to your crowd"
},
{
  category: "Jackets",
  default_price: "140.00",
  description: "The So Fatigues will wake you up and fit you in. This high energy camo will have you blending in to even the wildest surroundings.",
  id: 40346,
  name: "Camo Onesie",
  slogan: "Blend in to your crowd"
},
{
  category: "Jackets",
  default_price: "140.00",
  description: "The So Fatigues will wake you up and fit you in. This high energy camo will have you blending in to even the wildest surroundings.",
  id: 40347,
  name: "Camo Onesie",
  slogan: "Blend in to your crowd"
},
];

const testProduct = {
  category: "Jackets",
  default_price: "140.00",
  description: "The So Fatigues will wake you up and fit you in. This high energy camo will have you blending in to even the wildest surroundings.",
  id: 40344,
  name: "Camo Onesie",
  slogan: "Blend in to your crowd"
}

const testDetails = shallow(<ProductDetails
  updateStyle={jest.fn()}
  products={testList}
  id={40344}
  scrollReview={jest.fn()}
  calculateStars={jest.fn()}
  reviewsAvgScore={jest.fn()}
  currentProduct={testProduct}/>)

  test('Component should render properly and stay consistent', () => {
    expect(toJson(testDetails)).toMatchSnapshot();
  })

  test('Component should update properly', () => {


    const spy = jest.spyOn(ProductDetails.prototype, "componentDidUpdate")
    const wrapper = shallow(<ProductDetails
      updateStyle={jest.fn()}
      products={testList}
      id={40344}
      scrollReview={jest.fn()}
      calculateStars={jest.fn()}
      reviewsAvgScore={jest.fn()}
      currentProduct={testProduct}/>)
    wrapper.setProps({
      id: 240501})
    expect(spy).toHaveBeenCalled();
  })