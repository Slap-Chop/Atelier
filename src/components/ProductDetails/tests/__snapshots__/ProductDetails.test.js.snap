// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component should render properly and stay consistent 1`] = `
<div
  className="productDetailsBody"
>
  <div
    className="imageGalleryBody"
  >
    <ImageGallery
      id={0}
      style={Object {}}
    />
  </div>
  <div>
    <div
      className="productInfoBody"
    >
      <ProductInfo
        calculateStars={[MockFunction]}
        click={[Function]}
        currentProduct={
          Object {
            "category": "Jackets",
            "default_price": "140.00",
            "description": "The So Fatigues will wake you up and fit you in. This high energy camo will have you blending in to even the wildest surroundings.",
            "id": 40344,
            "name": "Camo Onesie",
            "slogan": "Blend in to your crowd",
          }
        }
        currentStyle={Object {}}
        id={40344}
        reviewsAvgScore={[MockFunction]}
        scrollReview={[MockFunction]}
        styles={Array []}
      />
      <Cart
        currentProduct={
          Object {
            "category": "Jackets",
            "default_price": "140.00",
            "description": "The So Fatigues will wake you up and fit you in. This high energy camo will have you blending in to even the wildest surroundings.",
            "id": 40344,
            "name": "Camo Onesie",
            "slogan": "Blend in to your crowd",
          }
        }
        currentStyle={Object {}}
      />
      <div
        className="productBorder"
      >
        <h5>
          Blend in to your crowd
        </h5>
        <p>
          The So Fatigues will wake you up and fit you in. This high energy camo will have you blending in to even the wildest surroundings.
        </p>
      </div>
    </div>
  </div>
</div>
`;
